File: 프로그래머스/영어 끝말잇기/지민/영어 끝말잇기.py
#Summer/Winter Coding(~2018) 

def solution(n, words):
    answer = []
    for i in range(1, len(words)):
        if words[i-1][-1] != words[i][0] or words[i] in words[:i] or len(words[i]) == 1:
            answer = [(i%n)+1, (i//n)+1]
            break
    if len(answer) == 0: answer = [0, 0]
    return answer

File: 프로그래머스/영어 끝말잇기/수경/영어 끝말잇기.cpp
#include <string>
#include <vector>
#include <iostream>
#include <algorithm>

using namespace std;

vector<int> solution(int n, vector<string> words) {
    
    vector<int> answer{0, 1};
    vector<string> v; string tmp = ""; int i= 0; 
    for(auto elem : words){
        i++; 
        if(i > n){ i %= n; answer[1]++; }    
         
        if(!v.empty() && (elem.size() == 1 || tmp[tmp.size()-1] != elem[0] || find(v.begin(), v.end(), elem) != v.end())){ 
            answer[0] = i; 
            return answer;
        }   
        
        v.push_back(elem); 
        tmp = elem;
    }
    
    return {0, 0};
}